### DISCUSSION TITLE
مقترح لمنصة لعرض المشاريع وسهولة إضافتها

### DISCUSSION THREAD

**Discussion by AbdelrahmanBayoumi** [mobadarah/tajmeeaton#11] [Ideas]
# 💡 الهدف:

تجميعة قدر المستطاع للمشاريع البرمجية التي تساهم في (اللغة العربية والإسلام).. 

<h3 align=center>وعرضها بطريقة واضحة وسهلة البحث</h3>

---

# طريقة مشاركة المشروع:

<div align=right>

- تسجيل الدخول في المنصة
- عمل طلب إضافة المشروع
- يتم مراجعة المشروع أولا (من المسؤولين) ثم الموافقة على نشره على المنصة

</div>

---

## 📌 Main Features

<details>
<summary>🏡 Home</summary>

- Search Box
    - search by `tag` || by `projectName`
- View Projects
    - View number of reacts in project
    - Author name
- Add Project Button
    - redirect user to login if not logged-in
  
</details>

<details>
<summary>💼 Project Home</summary>

1. Author
2. Published Date
3. Updated Date
4. Project Name
5. Project Description (String | HTML | Markdown)
6. URL
    1. GitHub Repo
    2. Website
7. Thumbnail / Logo
8. Tags[]
9. Reactions: int ❤️
10. Comments[] sort( old in top )
  
</details>

<details>
<summary>📝 Add Project</summary>

- Name
- Description (String | HTML | Markdown)
- URL
    1. GitHub Repo
    2. Website
- Thumbnail / Logo
- Tags[]
</details>

<details>
  <summary>👥 <b>Site users</b> ⇒ (user, moderator, admin)</summary>

1. admin ⇒ 
    1. can (add/delete) moderators
    2. all admin privileges
2. moderator ⇒ 
    1. can accept/reject project requests
    2. (display & delete & respond) to users reports
    3. all user privileges
3. user ⇒ can do:
    1. see projects
    2. add projects
    3. react to the project 
    4. comment to the project
    5. report project to admins
  
</details>

<details>
<summary>📋 Dashboard</summary>
  
- 👤 Owner View
    - All 👤 Admin View +
    - view list of users & admins
        - can upgrade user to admin
        - can downgrade admin to user
- 👤 Admin View
    - List of pending posts
        - can accept/reject project requests
    - List of Reports
        - (delete & respond) to users reports
  
</details>

---

## 📦 Models

<details>
  <summary>👤 User - المستخدم</summary>

1. Personal Data
2. Privileges (user, moderator, admin)
3. published projects[] (projectIDs)
 
</details>

<details>
  <summary>💼 Project - المشروع</summary>
  
1. Author ⇒ UserID
2. Published Date 
3. Updated Date
4. Project Name
5. Project Description (String | HTML | Markdown)
6. URL
    1. GitHub Repo
    2. Website
7. Thumbnail / Logo
8. Tags[] (TagIDs)
9. Reactions: int ❤️
10. Comments[] (commentIDs)
  
</details>

<details>
  <summary>💬 Comments - تعليقات</summary>

1. Comment ID
2. Author (UserID)
3. Body (String | HTML | Markdown)
4. Created Date
5. Updated Date
6. Reactions: int ❤️
  
</details>

---

## 💡 Future Features

- User can follow tags
- User can subscribe to tag to get projects updates to E-mail

**Comment by AbdelrahmanBayoumi** [comment]
## 📦 مقارنة بسيطة على بعض قواعد البيانات (المجانية) الممكن استخدامها

![image](https://user-images.githubusercontent.com/48678280/156671781-a25182bb-3524-4f91-8c92-191c17079e22.png)


### 📌 أظن أن أفضل حل هو `MongoDB Atlas` يليه `Firebase - Cloud Firestore`

### 🖼️ بالنسبة للصور أظن أن `Firebase Cloud Storage` تكفي جدا في البداية

**Reply by assawalhy** [reply]
جزاك الله خيرا

**Reply by AbdelrahmanBayoumi** [reply]
> جزاك الله خيرا

وإياكم ✨

**Comment by abderrahmaneMustapha** [comment]
السلام عليكم  اخي , لقد اعجبتني الفكرة كثيررا و اود ان اطلعك على موقع و مبادرة تحت اسم dzcode نقوم بالعمل عليها الان يمكن الاطلاع على [موقعنا ](https://www.dzcode.io/) بالاضافة الى رابط المشروع على [github](https://github.com/dzcode-io) , فموقعنا يقوم بنفس ما تريد القيام به الان و نحن نرحب باي مشاركة بيننا او اقتراح او مشروع نفيد به لغتنا و ديننا و بلداننا

**Reply by islameldigwi** [reply]
المنصة تبدو جميلة، تمنيت لو كانت بالعربية.

**Reply by assawalhy** [reply]
<div dir=rtl>

عذرا على تأخر الرد فقد كنت مشغولا في الفترة السابقة وأخرت الرد إلى أن أدرس مشروع dzcode وأقارن بينه وبين المنصة المطلوبة وكلما أردت الرد تهيّبت وأجلّته إلى أن أجد الوقت لدراسة الأكواد بشكل أفضل، لكن أظن أنه يجب علي أن أحسم الأمر 😅.

## مشروع dzcode

هذا تحليلي لأكواد dzcode وإن أخطأت فصوب لي:
1. استخدمت في الموقع React وتطبيقات الهواتف React native وهذا جيد إذ أنها أطر العمل التي أُطور بها لكن لو كان الموقع ب Nextjs لكان أفضل من ناحية الظهور وتحسين محركات البحث SEO
2. مدير الحالة Redux
3. استخدمت طريقة ال monorepo في تنظيم المشروع وتقسيمه باستعمال yarn + lerna
4. تستعمل firebase كقاعدة البيانات
5. تستعمل الصور من مصدرها ولا تخزنها في استضافة خاصة بك 👌🏼 
6. المشاريع المُدرجة هي المشاريع في data/models/projects ولكنها تُبنى في ملف واحد وهو list.c.json باستخدام `yarn generate:data`
7. تستعمل github api للحصول على المساهمات وال issues وغيرها من البيانات وهذا في حزمة api. ويحصل هذا كلما طلبنا أحد نقاط النهاية مثل <span dir=ltr>`/Contributors`</span>
8. .......................... 
9. وغيرها الكثير من خصائص dzcode

## أوجه الاختلاف

منصة dzcode تشبه بشكل كبير ما نريد إنشاءه لكن تنقصها مزايا مثل:
1. الإضافة والحذف والتعديل من الموقع والتطبيق دون الحاجة ل pull request
2. التعليث commenting والتصويت voting
3. عرض للمشاريع مقسمة في كل قسمه وتحت وسوم tags والبحث فيها. مقسمة مثلا كما في ال README [للتجميعة](https://github.com/mobadarah/tajmeeaton).
4. لم نفكر في صفحات Learn, Articles, and Contribute ولكنها فكرة جيدة ❤️ 
5. الموقع نريده أن يكون بالعربية لكن لا مشكلة من عولمته بأكثر من لغة لكن تكون العربية هي اللغة الأساسية في الموقع، ورأيت في الكود شيئا مخصصا لل i18n لكن ليس بالموقع إلا الإنجليزية


## هل نبدأ من الصفر؟

رأيت الجهد الذي بذل في dzcode، فجزاك الله خيرا، والبدء من الصفر في مشروع وبذل جهود جديدة وقد وُجدت وبذُلت فعلا ليس بقرار صحيح. فأرى أن ننشيء تشعّبا من مستودعكم وأن نبدأ في تشكيل المشروع ليلائم المنصة وهذا بالتعاون والمساهمة بيننا فالعمل الجماعي ومشاركة المجتمع community هو الوقود لمثل هذه المشاريع، فكما رأيت تأخرت في الرد بسبب المشاغل التي لا تنتهى 😢.

**Reply by islameldigwi** [reply]
لدئ سؤال، في ماذا تحتاج المنصة إلى تعليقات؟ يبدو لي إن كل مشروع عبارة عن مستودع، يمكن كتابة issues فيه مباشرة، فأظن لا حاجة في أن تكون المنصة وسيط في التعليقات؟ ما هو تخيلك للتعليقات؟

**Comment by islameldigwi** [comment]
لماذا يجب على المنصة توفير تسجيل الدخول؟ من ممكن عمل repo في المشاريع عبارة عن Markdown files ومن الممكن عمل API لها، أو استخدامها مباشرة ب vuepress على سبيل المثال. أستطيع العمل على هذا إن لم يكن يعمل أحدًا عليه الآن.

**Reply by AbdelrahmanBayoumi** [reply]
> لماذا يجب على المنصة توفير تسجيل الدخول؟

في تصوري للمنصة نحتاج إلى تسجيل الدخول:
- عند كل مشروع يكون جزء خاص للتعليقات على المشروع.
- وكانت هناك فكرة لإمكانية إضافة المشروع بدون الحاجة لحساب على GitHub، عن طريق المنصة.. ولكن هذا يتطلب وجود **`مراجعين`** لطلب إضافة المشروع حتى يكون الأمر دقيقًا ولا يتم إضافة أي مشروع عشوائي  [ISSUE #7](https://github.com/mobadarah/tajmeeaton/issues/7#issuecomment-986204990)

**Reply by AbdelrahmanBayoumi** [reply]
> من ممكن عمل repo في المشاريع عبارة عن Markdown files ومن الممكن عمل API لها

هكذا يكون إضافة المشاريع يتطلب حساب على GitHub ولا أظنها مشكلة كبيرة، كان من الأفضل بالطبع توفر هذه الإمكانية في المنصة ولكن الغرض الرئيسي هو عرض المشاريع **_بطريقة بسيطة وسهلة البحث_** ..

**Reply by islameldigwi** [reply]
ربما يمكنني استخدام notion api لجلب هذه الروابط، إن كانت على شكل database مقسمة بالتصنيفات مثلًا. بالنسبة لطلبات الإضافة، ربما يمكننا بناء form في notion كصفحة مثلًا، ويقوم القائمون على المنصة بمراجعة الطلبات بشكل دوري وقبولها او رفضها مع تعليق مثلًا.

**Reply by islameldigwi** [reply]
كنت أعمل على شيئاً خاص بي، ربما يمكننا توحيد المجهودات؟
<img width="1228" alt="Screenshot 2022-03-27 at 18 46 45" src="https://user-images.githubusercontent.com/49557793/160291821-e63b7b3b-ac13-4eac-a8b9-69edf56274a3.png">

**Reply by assawalhy** [reply]
<div dir=rtl>

> ربما يمكننا بناء form في notion كصفحة مثلًا، ويقوم القائمون على المنصة بمراجعة الطلبات بشكل دوري وقبولها او رفضها مع تعليق مثلًا.

أظنها فكرة جيدة ومجانية تماما ضمن الخيارات المجانية من قواعد البيانات التي اقترحها @AbdelrahmanBayoumi في الأعلى

**Reply by assawalhy** [reply]
<div dir=rtl>

أنا في حيرة من أمري من كثيرة الطرق المختلفة لإنجاز هذا المشروع بالإضافة لإمكانية التعاون مع @abderrahmaneMustapha بدلا من البدء من الصفر والإكمال على مشروع dzcode. 

أظن أن المخرج من دوامة الحيرة هذه هو التفكير في أفضل وأسهل وأسرع طريق يؤدي للهدف lومعه يمكننا التعاون وتوزيع الحمل علينا بدلا من أن يكون المشروع تحت رحمة شخص واحد يعمل عليه حين يفرغ، ييأس حينا ويعمل عليه حينا آخر. فكروا معي! 🤔

**Reply by islameldigwi** [reply]
لضمان الاستمرارية علينا اتاحة الوصول لأكثر من شخص، فكرة أن نستخدم طرق مجانية سهلة وبسيطة هو  أمر ممتاز. ينبغي أن نكتب جزءًا ما عن كيفية المساهمة وما هى العملية التي يتم بها قبول المشتركين على سبيل المثال.

**Reply by islameldigwi** [reply]
لا مانع لدي بالتعاون مع dzcode، لكن في الحقيقة أفضل لو نستخدم مشروع مفتوح المصدر بالفعل مثل mkdocs مثلًا والتعامل ببساطة مع README files ولا حاجة لقاعدة بيانات في النهاية، أو ربما notion إن احتجنا (يتم سحب المحتوى أثناء عملية ال build ويكون في النهاية static website. أردت أن أنافش أيضًا ما هى معايير إضافة مشروع برمجي (ليس للتجمعية) بل مشروع مفتوح، يعمل عليه فريق مبادرة، أود ضمهم مجموعة لأضمن استمرارية هذه المشاريع، أعتقد أن كمنا بتوزيق الجهود بطريقة جيدة يمكننا أن نصل إلى نتائج ممتارة. 

dzcode لديه قائمة جيدة بالمشاريع وبأسئلة واجابات يمكننا ترجمتها،  
من الجميل أن نتكون من بلدان مختلفة لبناء مجهود عربي واحد، أؤمن بأن التناغم الثقافي هذا مهم ومن الممكن أن نبدع أن نظمنا العمل. يمكننا بناء Board هنا على github ومناقشة كل ما نريد من خلال ال issues، ربما يمكننا عمل نظام تصويت مثلًا.

**Reply by islameldigwi** [reply]
يمكننا أن نأخذ منظمة [gnome](https://www.gnome.org/get-involved/) على سبيل المثال لنبني المجتمع الخاص بنا، نقوم أولًا بالتعرف على أنواع المهام التي لدينا. مثلًا، تصميم، برمجة، تعريب مصطلحات، ترجمة، التواصل مع المجتمع، وربما الفريق الأساسي الذي يضع الرؤية ويقود ال initiatives ومنها نبني المهام في الأقسام التي ذكرتها.

**Reply by AbdelrahmanBayoumi** [reply]
أعجبتني جدا فكرة الmarkdown files 🚀
وما فهمته أن إضافة المشروع ستكون من خلال notion forms مثلا.. (من الممكن استخدام google forms أيضا) وربما يكون للمشروع المضاف حالة `قيد المراجعة` مثلا حتى يتم الموافقة عليه
ونستخدم notion API في عرض المشاريع من notion database التي في حالتها أنه تمت الموافقة عليها..

❌ ولكن كيف سيكون التعديل ؟ سنعود لفكرتين أساسيتين: 
- إما أن تكون المشاريع على هيئة JSON او markdown files المهم أنها ستكون في مستودع ما ويكون التعديل عن طريق `pull requests`
- أو يكون هناك خادم وقاعدة بيانات يتم حفظ فيه المشروع والتعامل مع التعديل (ربما يكون المستخدم الذي سيقوم بالتعديل هو صاحب المشروع أو ناشره.. وربما يكون شخص رأى خطأ ما فقام بطلب تعديل للمشروع وعندها الوضع الصحيح أنه يجب مراجعة طلبات التعديل مثل مراجعة طلبات الإضافة)

**Reply by AbdelrahmanBayoumi** [reply]
### استكمالا لفكرة الmarkdown files 
<div align=right dir=rtl>

✅ أظن أنه من الأفضل والأسرع ( لعدم توفر متطوعين للعمل على جزء الخادم `back-end` والAPI ) أن يكون المشروع قائم على ال `pull requests` من ناحية إضافة وتعديل المشاريع

📌 ما رأيكم في أن يكون مصدر البيانات للمنصة هو ملف JSON مثل هذا:

</div>


```json
{
  "title": "تجميعةٌ",
  "descreption": "مستودع يحتوي على مجموعة من المشاريع (وليس حصرا لجميعها) للنهوض بالأمة، وبشكل رئيسي تلك المشاريع مفتوحة المصدر مثل لغات بالبرمجة ومعالجة اللغات الطبيعية م.ل.ط، بالإضافة لمشاريع متعلقة بالإسلام مثل تطبيقات القرآن والأذكار والأذان وغيرها.",
  "howToContribute": "يمكنك المساهمة بتوضيح خطأ لغوي أو اقتراح تحسينات أو إضافة أعمال ومشاريع أخرى تخدم الأمة، يمكنك فتح issue أو المشاركة في النقاشات.",
  "notes": [
    "لا بد أن نعي أن معظم هذه المشروعات تعتبر جهود فردية أو فريق بسيط له شؤون حياته الخاصة وتعلمه وعمله، ومعظمها بَطُؤ أو تَوقّف تطويره، فلا بد أن نتحمل مسؤولية دعم هذه المشروعات ودفعها للأمام، سواء أكان دعما ماديا أو مساهمة في التطوير",
    "فليكن هدفك الوحيد رضا الله، ولذا أنصح برخصة 'وقف' بدلا من رخص مثل GPL وال MIT، وغيرهما. كما يمكن إنشاء رخصة بها شروط مختلفة منطلقة من نفس المبدأ لمشاريعك المستقبلية إن شاء الله"
  ],
  "projects": [
    {
      "name": "Azkar - أذكار",
      "shortDescreption": "برنامج لسطح المكتب 💻 يساعدك على: معرفة مواقيت الصلاة 🕌، قراءة أذكار الصباح و المساء مع التذكير بها 🤲، ظهور إشعارات بذكر او دعاء عشوائي كل فترة محددة 💬.",
      "tags": ["Azkar", "PrayerTimes", "Desktop App"],
      "links": {
        "GitHub": "https://github.com/AbdelrahmanBayoumi/Azkar-App",
        "webstie": "https://azkar-site.web.app/"
      },
      "thumbnail": "https://azkar-site.web.app/assets/images/logo.png",
      "longDescreption": "/path/to/markdown/file/Azkar.md"
    },
    {
      "name": "Adhan",
      "shortDescreption": "High precision prayer time library",
      "tags": ["PrayerTimes"],
      "links": {
        "GitHub": "https://github.com/batoulapps/Adhan"
      },
      "thumbnail": "DEFAULT_THUMBNAIL",
      "longDescreption": "/path/to/markdown/file/Adhan.md"
    }
  ]
}

```
📌 هذا الملف يحتوي على المشاريع التي ستظهر (في تصوري) في الصفحة الرئيسية او عند البحث مثلا.. وعند الضغط على المشروع إذا كان يحتوي على `longDescreption` سيتم عرض الشرح/الوصف الخاص بالمشروع المتوفر في المستودع

#### إضافة مشروع:
تكون بإضافته في هذا الملف + إضافة ملف markdown آخر لشرح المشروع إذا توفر (اختياري)
#### تعديل مشروع:
تكون بتعديله من هذا الملف أو تعديل الوصف/الشرح الخاص بالمشروع من ملف الmarkdown الموجود بالمستودع

**Comment by assawalhy** [comment]
<div dir=rtl>


حسنا أظن أننا على وشك الانتهاء من التفكير والبدء في العمل على المنصة وإن شاء الله ننتهي منها قريبا.


## الإضافة واقتراح التعديل

> الإضافة والتعديل والحذف بـ PR أو من google form مثلا

أظن أن أغلب من سيتعاملون مع المنصة هم مطورون ومبرمجون وتعاملهم مع الجيتهاب ونظام طلب السحب PR ليس بصعب عليهم.

ويمكننا توفير google form أو notion form، للتسهيل على من يريد. مع وجود تفقدنا المستمر للمدخلات الجديدة منها.

## كيف يكون المشروع

> المشاريع تخزن في ملفات markdown

كل مشروع يكون على هيئة markdown مع صدر الملف front matter واستخدام مكتبة مثل [gray-matter](https://www.npmjs.com/package/gray-matter) لقراءة المعلومات منها:

```markdown
---
الاسم: تجميعةٌ
الوصف: تجميعة من المشاريع، وخصوصا مفتوحة المصدر، للنهوض باللغة العربية والأمة الإسلامية. 👨🏼‍💻👨🏼‍🏫👨🏼‍🔬🧕
الوسوم: 
  - قائمة-رائعة
  - تجميعة
  - مجتمع-مبادرة
الروابط:
  جيتهاب: https://github.com/mobadarah/tajmeeaton
  موقع: tajmeaaton.com
---

# تجميعةٌ

مشروع بدأ بكذا وانتهى بكذا ويعمل عليه فلان وفلان من الناس، وإلخ.....
```

## البحث

> البحث في الطرف الأمامي

مهما زاد عدد المشاريع فإنه لا يمكن أن يكون بالحجم الذي يحتاج طرفا خلفيا backend للبحث فيها، ولو فرضنا أن لدينا عشرة آلاف، 10000، مشروع فجعلهم في مصفوفة في الطرف الأمامي والبحث المباشرة لن يكون ذا أثر يشعر به المستخدم للمنصة بل سيعجبه سرعة البحث.

## التعليقات

توجد خدمات تقدم لنا نظام تعليق جاهز مثل 

- https://github.com/utterance/utterances
- https://disqus.com/

وأظن أنها كافية بالنسبة لنا

**Reply by assawalhy** [reply]
<div dir=rtl>

الخطوة الأولى وتعتبر ٣٠٪ تقريبا من العمل وهي [إنشاء نموذج أولي](https://twitter.com/muhammad2052001/status/1508940174365020160)، لكني مشغول جدا ولعلكم لاحظتم تأخر الرد، فلو كان لديكم خبرة بتصميم النماذج الأولية أو تعرفون من يساهم معنا بإنشائه سيكون خيرا وإسراعا في إنشاء المنصة.

**Reply by islameldigwi** [reply]
لماذا نحتاج للتعليقات؟ لازلت لا أفهم. أن كانت التجميعة عبارة عن روابط ووصف لها، فما غاية التعليقات؟ 
إن كان الهدف هو التواصل بين فريق العمل، فيمكننا التواصل على ديسكورد أو سلاك مثلًا. أن كان الهدف هو النقاشات مع المجتمع، فربما يمكننا استخدام github discussions. هناك أيضًا مجتمع حسوب.

**Reply by assawalhy** [reply]
@AbdelrahmanBayoumi @schehata @abderrahmaneMustapha 

<div dir=rtl>

**ما آراءكم فيما انتهينا إليه؟**

**Reply by islameldigwi** [reply]
بالنسبة لنسخة نموذجية أولى، قد قمت بعمل deployment للتصميم الذي كنت أعمل عليه، وهو يعتمد على ال markdown، يمكن للتجمعية أن تكون مجموعة من الملفات أو ملف واحد، لا مشكلة. يمكن حتى آن تكون مجموعة من ملفات ال yaml. ألقوا نظرة على الموقع، أخبروني ما رأيكم وكيف نتقدم للأمام من هذه النقطة؟

يمكنك تصفح الموقع من [هنا](https://mobadarah.github.io)
الكود موجود بمستودع جديد داخل المنظمة.

**Reply by islameldigwi** [reply]
لدى سؤال أخر، هل في خطتكم أن تكون التجميعة في موقع منفصل عن المبادرة؟ أشعر أنني أخطأت الفهم أو الصورة غير واضحة إن كان هناك موقعان أم موقع واحد (موقع واحد للتجميعة أو للمبادرة).

**Reply by assawalhy** [reply]
> لماذا نحتاج للتعليقات؟

لكي يعلق المتصفحون للمشاريع ومستخدمو المنصة بتحسين للوصف أو تعقيب على المشروع. _وليس الغرض تواصل فريق العمل فيما بينهم._

كان بالإمكان توجيه من يريد التعليق أن ينشيء issue ويقدم فيها تعقيبه أو ملاحظته لكن الهدف من المنصة التسهيل والتنظيم، والخدمات المجانية متاحة لتوفير نظام للتعليقات.

وليس الأمر لزاما أن نضيفها، الحكمة أن نبدأ بما له أولوية أكبر وبذلك يمكننا جعل توفير التعليقات في المنصة آخر ما نعمل عليه بعد تصميم الواجهة لعرض المشاريع والبحث أولا.

**Reply by islameldigwi** [reply]
> لكي يعلق المتصفحون للمشاريع ومستخدمو المنصة بتحسين للوصف أو تعقيب على المشروع. _وليس الغرض تواصل فريق العمل فيما بينهم._
تحسين الوصف يبدو كمهمة يمكن القيام بها من خلال github، إستخدام التعليقات بهذا الشكل سيكون عبئًا علينا، حيث نحتاج شخص أو أشخاص يقوموا بمراجعة التعليقات وتصفية التعديلات والقيام بتلك على تلك المشروعات من خلال التعليقات، ستكون عملية مرهقة وغير مرتبة. أعتقد آننا نحتاج إلى التفرقة بين أنواع المستخدمين. فلدينا مطوريين، يقوموا ببناء التطبيقات أو إضافة مشروعاتهم من خلال github. وهم قادرين على ذلك بسهولة، فلا داعي لتصميم أداة جديدة لذلك.

بالنسبة للمستخدمين غير التقنيين، أرى أن ال requirements غير واضحة، وصعب أن نقوم ببناء منتج بدون الأخذ بطلبات هذه الفئة من المستخدمين. أنصح بأن ننتظر حتى يكون لدينا هذا النوع من المتطوعين أو من المستخدمين ومن ثم نبني حسب الاحتياج. 

بشكل عام، في رأيي يمكننا أن نسير بطريقة lean، تدريجيا نبني جزء جزء، دون الحاجة لبناء منصة ضخمة من البداية، لن ننتهي منها ابدأ.

> 
> وليس الأمر لزاما أن نضيفها، الحكمة أن نبدأ بما له أولوية أكبر وبذلك يمكننا جعل توفير التعليقات في المنصة آخر ما نعمل عليه بعد تصميم الواجهة لعرض المشاريع والبحث أولا.

أتفق معك تمامًِا.

**Comment by islameldigwi** [comment]
إن كانت التجميعة عبارة عن موقع منفصل، فيمكن أن تكون بهذا الشكل.

<img width="1439" alt="Screenshot 2022-03-30 at 13 15 53" src="https://user-images.githubusercontent.com/49557793/160823064-2f5c065b-30aa-4cbb-b34a-12a09296e402.png">

**Comment by ramyahm2010** [comment]
تحية طيبة،
اطلعت على هذا النقاش الرائع بخصوص إنشاء منصة لعرض المشاريع التي تخدم اللغة العربية والإسلام، وأود أن أشارككم بمبادرة مشابهة أعمل عليها حاليًا وهي منصة "مرتكز" – وهي منصة عربية تهدف إلى تسليط الضوء على التطبيقات والأفكار التقنية المميزة في العالم العربي، مع التركيز على المشاريع الناشئة، مفتوحة المصدر، أو حتى التجريبية منها.

منصة مرتكز تتيح لأي مطوّر أو صاحب فكرة نشر مشروعه بسهولة، وتمكين الزوار من التفاعل من خلال التصويت والمشاركة. كما نسعى لتوفير واجهة استخدام مبسطة، وخالية من أي عناصر مشتتة أو إعلانات مزعجة، مع التزامنا الكامل بدعم المحتوى العربي والخصوصية.

يسعدني تواصلكم للإطلاع على المنصة من خلال الرابط:
🔗 https://murtakaz.com

وأتمنى أن تجدوا فيها ما يساهم في دعم هذا الهدف النبيل الذي طرحتموه هنا.

مع خالص التقدير لكم ولجهودكم 🌿
